test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -288,3 +266,100 @@ func TestVLANNetwork(t *testing.T) {
 	}
 	testSecondaryNetwork(t, networkTypeVLAN, pods)
 }
+
+func (data *testData) assignIP() error {
+	e2eTestData := data.e2eTestData
+	namespace := e2eTestData.GetTestNamespace()
+
+	for _, testPod := range data.pods {
+		_, err := e2eTestData.PodWaitFor(defaultTimeout, testPod.podName, namespace, func(pod *corev1.Pod) (bool, error) {
+			if pod.Status.Phase != corev1.PodRunning {
+				return false, nil
+			}
+			podIPs, err := data.listPodIPs(testPod)
+			if err != nil {
+				return false, err
+			}
+			ip, exists := podIPs["eth1"]
+			if !exists || ip == nil {
+				logs.Infof("IP not available for interface 'eth1' in Pod %s, retrying...", testPod.podName)
+				return false, nil
+			}
+
+			if err := aws.AssignIPToEC2ENI(testPod.secondaryENI, string(ip)); err != nil {
+				return false, err
+			}
+			logs.Infof("assigned private ip addresses %s to interface %s", ip, testPod.secondaryENI)
+			return true, nil
+		})
+		if err != nil {
+			return fmt.Errorf("error when waiting for secondary IPs for Pod %+v: %v", testPod, err)
+		}
+	}
+	return nil
+}
+
+func TestSRIOVNetwork(t *testing.T) {
+	e2eTestData, err := antreae2e.SetupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer antreae2e.TeardownTest(t, e2eTestData)
+
+	awsTestData, err := setupTest(e2eTestData)
+	if err != nil {
+		t.Fatalf("Error when setting up AWS test data: %v", err)
+	}
+
+	node0Name := antreae2e.NodeName(0)
+	node1Name := antreae2e.NodeName(1)
+	node0ENI, err := awsTestData.retrieveNodeENI(node0Name)
+	if err != nil {
+		t.Fatalf("Error when retrieve secondary NetworkInf ID of Node %s: %v", node0Name, err)
+	}
+	node1ENI, err := awsTestData.retrieveNodeENI(node1Name)
+	if err != nil {
+		t.Fatalf("Error when retrieve secondary NetworkInf ID of Node %s: %v", node1Name, err)
+	}
+	pods := []*testPodInfo{
+		{
+			podName:           "sriov-pod1",
+			nodeName:          node0Name,
+			interfaceNetworks: map[string]string{"eth1": "sriov-net1"},
+			secondaryENI:      node0ENI,
+		},
+		{
+			podName:           "sriov-pod2",
+			nodeName:          node1Name,
+			interfaceNetworks: map[string]string{"eth1": "sriov-net1"},
+			secondaryENI:      node1ENI,
+		},
+	}
+
+	testData := &testData{e2eTestData: e2eTestData, networkType: networkTypeSriov, pods: pods}
+
+	configJSON := `{
+		"cniVersion": "0.3.0",
+		"type": "antrea",
+		"networkType": "sriov",
+		"ipam": {
+			"type": "antrea",
+			"ippools": ["pool1"]
+		}
+	}`
+	err = awsTestData.createNetworkAttachmentDefinition(e2eTestData.GetTestNamespace(), "sriov-net1", configJSON)
```

Comment:
please revisit my earlier comment about creating the NetworkAttachmentDefinition from `sriov-secondary-networks.yml` and removing `createNetworkAttachmentDefinition`. I don't see any reason why this would work for the VLAN tests and not for the SRIOV tests. It should be possible to have the NetworkAttachmentDefinition in the default Namespace and the Pods in the test Namespace. This is what we have for the VLAN e2e tests, and the `"k8s.v1.cni.cncf.io/networks"` annotation should work the same independently of the network type.

Reply:
Updated.